cmd/nfd-topology-updater/main.go

```diff
@@ -92,9 +93,15 @@ func main() {
 	tmPolicy := string(topologypolicy.DetectTopologyPolicy(klConfig.TopologyManagerPolicy, klConfig.TopologyManagerScope))
 	klog.Infof("detected kubelet Topology Manager policy %q", tmPolicy)
 
-	// Get new TopologyUpdater instance
-	instance := topology.NewTopologyUpdater(*args, *resourcemonitorArgs, tmPolicy)
+	eventChan := make(chan notifier.Info)
+	ntf, err := notifier.New(resourcemonitorArgs.SleepInterval, eventChan, resourcemonitorArgs.KubeletDirPath)
+	if err != nil {
+		klog.Exit(err)
+	}
+	go ntf.Run()
```

Comment:
As I mentioned in the other comment, how about starting this in `topology.NewTopologyUpdater()`, similar to the resourcescanner. Would be more aligned with how things currently work. WDYT? Maybe it would be cleaner to start all these goroutines from `main()`, I'm not sure 🤨

Reply:
I agree, this notifier is only used by `TopologyUpdater`, so it's initialization and spinning it's goroutine can be hide within `NewTopologyUpdater`.

Reply:
Done